src/network/modules/tokens/types.ts

```diff
@@ -0,0 +1,56 @@
+import { Address } from "../../../identity"
+import { NetworkModule } from "../types"
+
+type LedgerAmount = BigInt
+
+export interface TokenInfoSummary {
+  name: string
+  symbol: string
+  precision: number
+}
+
+export interface TokenInfoSupply {
+  total: LedgerAmount
+  circulating: LedgerAmount
+  maximum?: LedgerAmount
+}
+
+export interface TokenInfo {
+  address: Address
+  summary: TokenInfoSummary
+  supply: TokenInfoSupply
+  owner?: Address
+}
+
+export interface TokensInfoParam {
+  address: Address
```

Comment:
Missing attribute-related index

```
    ; Query specific information. If missing, all information will be returned
    ; if possible (if it fits in response size) or error.
    ? 1 => [ + attribute-related-index ],
```

Reply:
Yeah, I don't really understand how attribute related indices work. Like, if I wanted to remove the visual logo for a token, do I do something like this?
```
tokens.removeExtendedInfo({
  0: tokenAddress,
  1: [[11,1]]
})
```
i.e. Attribute 11 (tokens) and then id 1 (visual image)? I'm not 100% sure how to capture this in TypeScript, as object keys have to be either strings or numbers. I might just serialize as strings.

Reply:
I took a shot.

Reply:
At the moment, `Memo` is `0` and `VisualLogo` is `1`, so you need to do

```
tokens.removeExtendedInfo({
  0: tokenAddress,
  1: [1]
})
```

to remove a `VisualLogo`.

See https://github.com/liftedinit/many-rs/pull/223/files#diff-ebe0beb42f021d0a8d3d08aacb86a8aad7cb3be2949b17b4b9169b4b6ee45adfR12-R23

Reply:
> I'm not 100% sure how to capture this in TypeScript, as object keys have to be either strings or numbers.

_Object_ keys have to be a string (numbers are just stringified), but `Map` objects don't have to have string keys. e.g. (in pure JavaScript):

<img width="922" alt="image" src="https://user-images.githubusercontent.com/681969/204612754-b1a10e3d-0733-4b5c-9ab3-72855e92e95f.png">

Now the problem happens when trying to read from the map as the array is an object (which are references in JS) and not a scalar (so `{} !== {}`):

<img width="922" alt="image" src="https://user-images.githubusercontent.com/681969/204612893-32909627-78b3-4987-9878-1913160deb8d.png">

What you can do instead is using iterators and verify each individually. You should also implement a specific type of AttributeRelatedIndex (which is what we do in Rust).

<img width="922" alt="image" src="https://user-images.githubusercontent.com/681969/204614336-33100dd6-54e1-46be-887a-76a032cfe71b.png">

It's not perfect but it should be performant _enough_.

A good reference: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map (with a comparison of maps vs objects)